packages/babel-types/src/clone/cloneNode.ts

```diff
@@ -29,6 +29,15 @@ export default function cloneNode<T extends t.Node>(
   node: T,
   deep: boolean = true,
   withoutLoc: boolean = false,
+): T {
+  return cloneNodeInternal(node, deep, withoutLoc, new Map());
```

Comment:
Should it be a WeakMap? If a comment node has been removed from AST, I don't think the cache should prevent it from being GC'ed.

Reply:
This `Map` will be destroyed after the function returns.

Because a comment in ast will be referenced by the two expressions before and after respectively, and the generator judges whether it is the same by comparing the comment object instance.
After deep cloning, it will become two different comment objects, resulting in repeated generation.

Use comment caching to avoid duplicate comments that are deeply cloned together.
If it is to be more perfect, it needs to be refactored in the future.